plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -88,19 +88,19 @@ consByteString :: Integer -> BuiltinByteString -> BuiltinByteString
 consByteString n bs = BI.consByteString (toBuiltin n) bs
 
 {-# INLINABLE sliceByteString #-}
--- | Returns the substring of a 'ByteString'.
+-- | Returns the substring of a 'ByteString' from index 'start' of length 'n'.
 sliceByteString :: Integer -> Integer -> BuiltinByteString -> BuiltinByteString
-sliceByteString from to bs = BI.sliceByteString (toBuiltin from) (toBuiltin to) bs
+sliceByteString start n bs = BI.sliceByteString (toBuiltin start) (toBuiltin n) bs
 
 {-# INLINABLE takeByteString #-}
 -- | Returns the n length prefix of a 'ByteString'.
 takeByteString :: Integer -> BuiltinByteString -> BuiltinByteString
-takeByteString n bs = BI.sliceByteString 0 (toBuiltin n - 1) bs
+takeByteString n bs = BI.sliceByteString 0 (toBuiltin n) bs
 
 {-# INLINABLE dropByteString #-}
 -- | Returns the suffix of a 'ByteString' after n elements.
 dropByteString :: Integer -> BuiltinByteString -> BuiltinByteString
-dropByteString n bs = BI.sliceByteString (toBuiltin n) (BI.lengthOfByteString bs - 1) bs
+dropByteString n bs = BI.sliceByteString (toBuiltin n) (BI.lengthOfByteString bs - n) bs
```

Comment:
`minusInteger`

Reply:
I think this will get the wrong `-` otherwise

Reply:
Alternatively, maybe we should define these in `PlutusTx.Prelude`, where the right `-` will be in scope, and these aren't builtins any more anyway.

Reply:
take and drop, that is

Reply:
Good catch!

It's my second time of using the wrong operation. Maybe it's worth to add `NoImplicitPrelude` everywhere and use a special module `AllowedPrelude`? 🤔